awx/main/migrations/0124_event_partitions.py

```diff
@@ -0,0 +1,112 @@
+# Generated by Django 2.2.8 on 2020-02-21 16:31
+
+from datetime import datetime
+
+from django.db import migrations, models, connection
+from django.utils.timezone import now
+
+from awx.main.utils.common import create_partition
+
+
+def migrate_event_data(apps, schema_editor):
+    # see: https://github.com/ansible/awx/issues/9039
+    #
+    # the goal of this function is to:
+    # - [ ] create a parent partition table, main_jobevent_parent
+    # - [ ] .. with a single partition
+    # - [ ] .. that includes all existing job events
+    #
+    # the new main_jobevent_parent table should have a new
+    # denormalized column, job_created, this is used as a
+    # basis for partitioning job event rows
+    #
+    # The initial partion will be a unique case. After
+    # the migration is completed, awx should create
+    # new partitions on an hourly basis, as needed.
+    # All events for a given job should be placed in
+    # a partition based on the job's _created time_.
+
+    # Only partitioning main_jobevent on first pass
+    #
+    #for tblname in (
```

Comment:
@jladdjr before you continue onward with with problems related to analytics, job event cleanup, and large-scale migrations, it might be good to pause and update this PR with some changes that introduce partitions for *all* event types - looking at the layout of this code and the comments here, I think you're pretty close 😄, and having that is probably a first pass of what QE and perfscale folks need to actually stand up and test this from a functional/correctness perspective.